src/bot.coffee

```diff
@@ -200,6 +200,8 @@ class SlackBot extends Adapter
     # NOTE: should rate limit errors also bubble up?
     if error.code isnt -1
       @robot.emit "error", error
+    else
+      @robot.emit "slack-rate-limit"
```

Comment:
it took quite a bit of detective work, but it turns out that we don't know of any code that triggers the `"error"` event on the `RtmClient`, which is the only code path that results in this method ever getting called.

it seems like the comparison of `code` to `-1` is remnants from the v3.x series of `hubot-slack`, which used the v1.x series of `@slack/client`, which would have emitted a `"error"` event.

therefore, i don't think this code is worth merging, because it would just become even more misleading.